build/simp-utils.spec

```diff
@@ -1,7 +1,50 @@
+%{lua:
+--
+-- When buildng the RPM, declare macro 'pup_module_info_dir' with the path to
+-- the top-level project directory.  This directory should contain the
+-- following items:
+--   * 'build' directory
+--   * 'README.md' file
+--
+package_release = '1'
+local potential_src_dirs = {
+  rpm.expand('%{pup_module_info_dir}'),
+  rpm.expand('%{_sourcedir}'),
+  posix.getcwd(),
+}
+local src_dir = "\0"
+
+for _k,dir in pairs(potential_src_dirs) do
+  if (posix.stat(dir .. '/build', 'type') == 'directory') and (posix.stat(dir .. '/README.md', 'type') == 'regular') then
+    src_dir = dir
+    break
+  end
+end
+
+if src_dir == "\0" then
+  error(
+    "FATAL: Cannot determine RPM project's src_dir!\n" ..
+    "\t* Paths tried: '" .. table.concat(potential_src_dirs, "', '") .."\n" ..
+    "\t* PROTIP: Declare macro %pup_module_info_dir\n"
+  )
+end
+
+-- Snag the RPM-specific items out of the 'build/rpm_metadata' directory
+rel_file = (io.open(src_dir .. "/build/rpm_metadata/release", "r") or io.open(src_dir .. "/release", "r"))
+if rel_file then
+  for line in rel_file:lines() do
+    if not (line:match("^%s*#") or line:match("^%s*$")) then
+      package_release = line
+      break
+    end
+  end
+end
+}
+
 Summary: SIMP Utils
 Name: simp-utils
 Version: 6.5.1
-Release: 1
+Release: %{lua: print(package_release)}%{?dist}
```

Comment:
Nothing in the spec file is distribution specific, so dist is not actually required.  Are we adding dist to all our RPMs, regardless?

Reply:
We'd talked about it, but certainly don't have to.   I'll take it out.